can/interfaces/socketcan/socketcan.py

```diff
@@ -473,12 +505,39 @@ def send(self, msg, timeout=None):
             if not ready_send_sockets:
                 raise can.CanError("Timeout while sending")
 
+        data = build_can_frame(msg)
         try:
-            self.socket.sendall(build_can_frame(msg))
+            if self.channel == "" and HAS_NATIVE_SUPPORT and msg.channel:
+                self.socket.sendto(data, (msg.channel, ))
+            else:
+                self.socket.send(data)
```

Comment:
Shouldn't we use `sendall()`, since `send()` might not transmit everything?

Reply:
The default behaviour for all other interfaces and previously for socketcan is to not block on `send()`. If it can't be transmitted, it should be raised as an exception and the application can decide if it wants to try again, move on to the next message or signal as an error. The most common scenario is that something is wrong with the bus and no nodes are ACK:ing your messages. If we would be using `sendall` then the application will just freeze.

I added the `timeout` argument for making `send` blocking, but usually you want to give up after a certain time. One problem with my change will arise if it is possible to queue only part of a message. Then we want to go back to `select` and loop until all bytes are sent, but only for the timeout period. I'll give that a go.